File: codewars/Valid Parentheses/Valid Parentheses.js
function validParentheses(parens) {
    let openingBracket = []
    let closingCheker = {
        ')' : '('
    }
    for (let i = 0; i < parens.length; i++) {
        letter = parens[i];
        if (letter == '(') {
            openingBracket.push(letter);
        } else if (letter == ')') {
            if (openingBracket.length == 0 || openingBracket.pop() !== closingCheker[letter]){
                return false;
            }
        }
    }
    return openingBracket.length == 0;
  }

File: codewars/Valid Parentheses/Valid_Parentheses.c
#include <stdbool.h>

bool validParentheses(const char *str_in) {
    int p = 0, i = 0;
    bool pr = true;
    while (str_in[i] != '\0') {
        if (str_in[i] == '(') p++;
        if (str_in[i] == ')') p--;
        if (p < 0) {
            pr = false;
            break;
        }
        i++;
    }
    if (p != 0) pr = false;
    return pr;
}
